homeassistant/components/smartthings/sensor.py

```diff
@@ -0,0 +1,219 @@
+            DEVICE_CLASS_TEMPERATURE)],
+    'thermostatMode': [
+        Map('thermostatMode', "Thermostat Mode", None, None)],
+    'thermostatOperatingState': [
+        Map('thermostatOperatingState', "Thermostat Operating State",
+            None, None)],
+    'thermostatSetpoint': [
+        Map('thermostatSetpoint', "Thermostat Setpoint", TEMP_CELSIUS,
+            DEVICE_CLASS_TEMPERATURE)],
+    'tvChannel': [
+        Map('tvChannel', "Tv Channel", None, None)],
+    'tvocMeasurement': [
+        Map('tvocLevel', "Tvoc Measurement", 'ppm', None)],
+    'ultravioletIndex': [
+        Map('ultravioletIndex', "Ultraviolet Index", None, None)],
+    'voltageMeasurement': [
+        Map('voltage', "Voltage Measurement", 'V', None)],
+    'washerMode': [
+        Map('washerMode', "Washer Mode", None, None)],
+    'washerOperatingState': [
+        Map('machineState', "Washer Machine State", None, None),
+        Map('washerJobState', "Washer Job State", None, None),
+        Map('completionTime', "Washer Completion Time", None,
+            DEVICE_CLASS_TIMESTAMP)],
+    'windowShade': [
+        Map('windowShade', 'Window Shade', None, None)]
+}
+
+UNITS = {
+    'C': TEMP_CELSIUS,
+    'F': TEMP_FAHRENHEIT
+}
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Platform uses config entry setup."""
+    pass
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add binary sensors for a config entry."""
+    broker = hass.data[DOMAIN][DATA_BROKERS][config_entry.entry_id]
+    sensors = []
+    for device in broker.devices.values():
+        for capability, maps in CAPABILITY_TO_SENSORS.items():
+            if capability in device.capabilities:
+                sensors.extend([
+                    SmartThingsSensor(
+                        device, m.attribute, m.name, m.default_unit,
+                        m.device_class)
+                    for m in maps])
+    async_add_entities(sensors)
+
+
+class SmartThingsSensor(SmartThingsEntity):
+    """Define a SmartThings Binary Sensor."""
+
+    def __init__(self, device, attribute: str, name: str,
+                 default_unit: str, device_class: str):
+        """Init the class."""
+        super().__init__(device)
+        self._attribute = attribute
+        self._name = name
+        self._device_class = device_class
+        self._default_unit = default_unit
+
+    @property
+    def name(self) -> str:
+        """Return the name of the binary sensor."""
+        return '{} {}'.format(self._device.label, self._name)
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return '{}.{}'.format(self._device.device_id, self._attribute)
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._device.status.attributes[self._attribute].value \
+            or STATE_UNKNOWN
```

Comment:
We should return `None`. HA internally will publish `STATE_UNKNOWN`

Reply:
Cleaned up!